### DISCUSSION TITLE
Fetch local audio file (file:///...) from within react-native-webview

### DISCUSSION THREAD

**Discussion by Patrick-Sawyer** [react-native-webview/react-native-webview#1805] [General]
I am trying to get an audio file (mp3, wav) as an audioBuffer in a react native app. Currently my approach is to use the web audio API within the react-native-webview component, and send data from there to the main react native app (ultimately to make an audio waveform). This works really well when I fetch an mp3 from a remote URL, but fails when I try to access a file on an android - e.g. file:///........

I am using the "injectedJavaScript" prop to inject the JavaScript below into my react-native-webview component.

Any help on this would be amazing, I have lost days to this. Tried react-native-audio-waveform but it is far too slow.

const injectedJavaScript = `

     //create audio context

     window.AudioContext = window.AudioContext || 
     window.webkitAudioContext;

     const audioContext = new AudioContext();

     //function to get arrayBuffer and convert to audioBuffer

     const getBuffer = url => {
       fetch(url)
         .then(response => response.arrayBuffer())
         .then(arrayBuffer => audioContext.decodeAudioData(arrayBuffer))
         .then(audioBuffer => filterData(audioBuffer))
         .catch(error => window.ReactNativeWebView.postMessage(error))
     };
     
     //function below filters data to a smaller array
     
     const filterData = audioBuffer => {
       console.log("filterData called");
       const rawData = audioBuffer.getChannelData(0);
       const samples = 10000; 
       const blockSize = Math.floor(rawData.length / samples);
       let filterData = [];
       for (let i = 0; i < samples; i++) {
        let segment = blockSize * i;
        let value = rawData[segment];
        filterData.push(value);
       }

       //sends the data back to the react native app
       window.ReactNativeWebView.postMessage(JSON.stringify(filterData));
     }

     //call the function with a local mp3 file 
     //all code works fine when this function is called with a remote URL e.g. "http://wwww.blahblah"

     getBuffer("file:///storage/emulated/0/Music/blahblahblah.mp3")

     `;

**Comment by anzorb** [comment]
@Patrick-Sawyer have you figured this out?

**Comment by danfri86** [comment]
I'm trying the exact same thing but am not able to fetch my audio file with the file:/// prefix.
Did you find a solution @Patrick-Sawyer? It's been some time so you might have forgot, but I'll give it a shot!
